odoo/fields.py

```diff
@@ -4595,23 +4600,26 @@ def browse(ids):
                     if command[0] == Command.CREATE:
                         for record in recs:
                             line = comodel.new(command[2], ref=command[1])
-                            line[inverse] = record
+                            to_inverse[record].add(line.id)
                     elif command[0] == Command.UPDATE:
                         browse([command[1]]).update(command[2])
                     elif command[0] == Command.DELETE:
-                        browse([command[1]])[inverse] = False
+                        to_inverse[False].add(command[1])
                     elif command[0] == Command.UNLINK:
-                        browse([command[1]])[inverse] = False
+                        to_inverse[False].add(command[1])
                     elif command[0] == Command.LINK:
-                        browse([command[1]])[inverse] = recs[-1]
+                        to_inverse[recs[-1]].add(command[1])
                     elif command[0] == Command.CLEAR:
+                        flush_to_inverse()
                         cache.update(recs, self, itertools.repeat(()))
                     elif command[0] == Command.SET:
+                        flush_to_inverse()
                         # assign the given lines to the last record only
                         cache.update(recs, self, itertools.repeat(()))
                         last, lines = recs[-1], browse(command[2])
                         cache.set(last, self, lines._ids)
                         cache.update(lines, inverse_field, itertools.repeat(last.id))
```

Comment:
I suggest this alternative, using my proposal for `to_assign` above.
```suggestion
                        flush()  # to apply the command to the result of former commands
                        # assign the given lines to the last record only
                        record = recs[-1]
                        line_ids = set(command[2])
                        lines_to_unlink = record[self.name].filtered(lamdba line: line._origin.id in line_ids)
                        lines_to_unlink[inverse] = False
                        to_assign.update(dict.from_keys(command[2], record)
```
And I suggest to adapt this one to `Command.CLEAR`, too.  The only difference is that `line_ids` is always empty in the case of `Command.CLEAR`.

Note about the way to unlink lines above.  You cannot unlink the lines that are discarded out by `Command.CLEAR` and `Command.SET` with `to_assign` (or `to_inverse`), because those dicts only contain _real_ line ids, while the field value may contain _new_ records without origin.